com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -1163,8 +1256,31 @@ private void SetStateServerRpc(Vector3 pos, Quaternion rot, Vector3 scale, bool
             TryCommitTransform(transform, m_CachedNetworkManager.LocalTime.Time);
         }
 
-        // todo: this is currently in update, to be able to catch any transform changes. A FixedUpdate mode could be added to be less intense, but it'd be
-        // conditional to users only making transform update changes in FixedUpdate.
+        /// <summary>
+        /// If the instance is authority it will attempt to update the current tick
+        /// network state and returns true.
+        /// If it is not authority it exits early returning false.
+        /// </summary>
+        /// <param name="transformSource">transform to be updated</param>
+        private bool TryUpdateAuthority(Transform transformSource)
```

Comment:
This name is confusing - it's not a method that Updates the Authority. It's a method that Tries to Commit Transform, and Also Reset Sync State.

I don't fully grok the replicated & local authoritative dirty disparity, but it'd be nice to have a more representative name for this method (or just split It into 2 methods... there's only 2 call sites so it wouldn't be the end of the world)

Reply:
Well, I needed to have a single place where authority updates and (to me) it was cleaner this way.
The big reason was that I was starting to have to maintain two sets of code... so perhaps I will just rename it to "UpdateAuthoritativeState", migrate the CanCommitToTransform check back into both methods, and keep the rest of the code/structure.

It doesn't just commit the transform...  :)
It also checks for the NetworkVariable to have been updated, creates a copy of the the local authoritative state, and then resets the local authoritative state.